src/tests/tracing/runtimeeventsource/RuntimeEventSourceTest.cs

```diff
@@ -1,114 +1,94 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System;
-using System.IO;
+#if USE_MDT_EVENTSOURCE
+using Microsoft.Diagnostics.Tracing;
+#else
 using System.Diagnostics.Tracing;
-using System.Runtime.CompilerServices;
+#endif
+using System;
+using System.Collections.Generic;
 using System.Threading;
-using Tracing.Tests.Common;
+using System.Threading.Tasks;
+using System.Diagnostics;
 
-namespace Tracing.Tests
+AppContext.SetSwitch("appContextSwitch", true);
+AppDomain.CurrentDomain.SetData("appContextBoolData", true); // Not loggeed, bool key
+AppDomain.CurrentDomain.SetData("appContextBoolAsStringData", "true");
+AppDomain.CurrentDomain.SetData("appContextStringData", "myString"); // Not logged, string does not parse as bool
+AppDomain.CurrentDomain.SetData("appContextSwitch", "false"); // should not override the SetSwitch above
+
+// Create an EventListener.
+using (var myListener = new RuntimeEventListener())
 {
-    public sealed class RuntimeEventSourceTest
+    await Task.Delay(10);
```

Comment:
Would you increase the timeout or do something else? In theory, when we create the runtime listener it should immediately start dumping the AppContext events, so I don't think the delay should even be necessary. But I don't know the internals of EventSource.

Reply:
Ah, I think you may be right about the delay not being necessary, since the AppContext events are dumped synchronously on event source enable. (@josalem?)

My mind was on other tracing scenarios that are capturing events for some action not the current state at enable. In those cases, I've seen (and written) tests spin waiting until we get some expected event or events.

Reply:
Agreed, you should not need the `Task.Delay` call.  All writes to an `EventListener` are synchronous on the thread that does the write.